R/reduce.R

```diff
@@ -170,25 +171,43 @@ seq_len2 <- function(start, end) {
 #'     geom_line(aes(color = simulation)) +
 #'     ggtitle("Simulations of a random walk with drift")
 #' }
-accumulate <- function(.x, .f, ..., .init) {
+accumulate <- function(.x, .f, ..., .init, .keep_names = FALSE) {
   .f <- as_mapper(.f, ...)
 
   f <- function(x, y) {
     .f(x, y, ...)
   }
+  
+  # Stop early so costly Reduce is called only when everything is fine
+  if (!is.logical(.keep_names) || length(.keep_names) != 1 || is.na(.keep_names)) {
+    rlang::abort("`.keep_names` must be TRUE or FALSE")
```

Comment:
You could use `is_logical(.keep_names, 1L) && !is.na(.keep_names)`. Also we don't prefix rlang functions with a namespace.

Reply:
We're getting rid of that check for this PR, so this is advice for future reference.

Reply:
Thanks for the advice.